unit_tests/system/profiler_test.cpp

```diff
@@ -9,69 +9,70 @@
 #include <cstdlib>
 #include <sstream>
 
-
-
 #define TEST_USE_MPI
+#define TEST_USE_PETSC
 #include <flow_gtest_mpi.hh>
 
-
-
+/**
+ * Fixture class for testing Profiler protected members
+ */
+class ProfilerTest: public testing::Test {
+    public:
+        void test_str_hash();
```

Comment:
The main problem is that **instance** of Profiler already exists and is static meaning no other class have access to its members if they are protected.
I did not manage to find any other way to solve this issue. There are couple of ideas ([[1]](http://stackoverflow.com/a/30075309/2616048), [[2]](http://stackoverflow.com/a/3676936/2616048) and [[3]](https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#testing-private-code))

But the problem with using only `TEST_F` is following (taken from GoogleTest docs)

> Private class members are only accessible from within the class or by friends. To access a class' private members, you can declare your test fixture as a friend to the class and define accessors in your fixture. Tests using the fixture can then access the private members of your production class via the accessors in the fixture. **Note that even though your fixture is a friend to your production class, your tests are not automatically friends to it, as they are technically defined in sub-classes of the fixture.**

So we need to have testing code in fixture class methods, otherwise we have no access to it (and inheritance wont help us). Example of ProfilerTest fixture class:
```c++
class ProfilerTest: public testing::Test, public Profiler {
};

TEST_F(ProfilerTest, test_memory_propagation) {
    cout << Profiler::instance()->timers_.size() << endl;
}
```
compilation error:
```c++
sys_profiler.hh:730:53: error: ‘std::vector Profiler::timers_’ is protected


```